streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -66,14 +66,14 @@ V deserializeValue(final String topic, final Headers headers, final byte[] data)
     @SuppressWarnings("unchecked")
     @Override
     public void init(final InternalProcessorContext context) {
-        super.init(context);
-        this.context = context;
         processAtSourceSensor = ProcessorNodeMetrics.processorAtSourceSensorOrForwardSensor(
             Thread.currentThread().getName(),
             context.taskId().toString(),
             context.currentNode().name(),
             context.metrics()
         );
+        super.init(context);
+        this.context = context;
 
```

Comment:
This is the fix. If the parent class `ProcessorNode` is initialised before the the actual sensor node an empty sensor without parent is created in `ProcessorNode#init()` due to backwards compatibility. Thus, on source nodes we need to first call the initialisation of the source node before we call the initialisation in the parent class.

Reply:
I think it's worth to add a comment about this (even if we have a test now)

Reply:
Done